src/Microphysics2M.jl

```diff
@@ -488,8 +488,11 @@ function radar_reflectivity(
     N_liq *= FT(1e-9)
     N_rai *= FT(1e-9)
 
-    xc = (N_liq == 0) ? FT(0) : ((q_liq * ρ_air) / N_liq)
-    xr = (N_rai == 0) ? FT(0) : ((q_rai * ρ_air) / N_rai)
+    q_liq = (q_liq < maximum(q_liq) * 1e-8) ? FT(0) : q_liq
+    q_rai = (q_rai < maximum(q_rai) * 1e-8) ? FT(0) : q_rai
+
+    xc = (N_liq < 1e-4) ? FT(0) : ((q_liq * ρ_air) / N_liq)
```

Comment:
I'm wondering if with those additional checks for q and N not being too small, we could get rid of the unit conversions and keep the whole computation in base SI?

Reply:
I analyzed the function and this is what I think. As we know, the problem lies in the computation of x; specifically, when using the SI units, q and ρ are very small, and N is very large, so x can become very large. This causes a blow up for B, and a subsequent infinite value for A. 
To solve this, I was able to remove the change of unit for the density, but I think that the change of units for N (from m^-3 to mm^-3) is needed. 
I specified it with a comment above it. Let me know what you think!